File: FragmentsApp/app/src/main/java/com/example/fragmentsapp/firstFragment.java
package com.example.fragmentsapp;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;


public class firstFragment extends Fragment {
    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        Toast.makeText(context, "onAttatch() is called", Toast.LENGTH_LONG).show();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Toast.makeText(getActivity(), "onCreate() is called", Toast.LENGTH_LONG).show();
    }

    @Override
    public void onResume() {
        super.onResume();
        Toast.makeText(getActivity(), "onResume() is called", Toast.LENGTH_LONG).show();

    }

    @Nullable
    @Override
    // most important method
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        // inflater - convert XML file into a view object in memory
        // inflate - create view from XML layout resource
        View view = inflater.inflate(R.layout.fragment_first,container,false);

        Button firstBtn = view.findViewById(R.id.btn_frag1);
        TextView text = view.findViewById(R.id.textView);

        firstBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getActivity(), "Welcome to the First fragment", Toast.LENGTH_SHORT).show();
            }
        });
        return view;
    }
}

File: FragmentDataSharing/app/src/main/java/com/tiramisu/fragmentdatasharing/FirstFragment.kt
package com.tiramisu.fragmentdatasharing

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction

class FirstFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view : View = inflater.inflate(R.layout.fragment_first, container, false)

        val name : EditText = view.findViewById(R.id.editTextName)
        val send : Button = view.findViewById(R.id.buttonSend)
        send.setOnClickListener {
            val userName = name.text.toString()
            val bundle = Bundle()
            bundle.putString("username", userName)
            val secondFragment = SecondFragment()
            secondFragment.arguments = bundle

            val fm : FragmentManager = requireActivity().supportFragmentManager
            val ft : FragmentTransaction = fm.beginTransaction()
            ft.replace(R.id.frame, secondFragment)
            ft.commit()
        }
        return view
    }
}